substrate/client/rpc-spec-v2/src/chain_head/chain_head_follow.rs

```diff
@@ -705,71 +710,109 @@ where
 	async fn submit_events<EventStream>(
 		&mut self,
 		startup_point: &StartupPoint<Block>,
-		mut stream: EventStream,
+		stream: EventStream,
 		sink: Subscription,
 		rx_stop: oneshot::Receiver<()>,
 	) -> Result<(), SubscriptionManagementError>
 	where
-		EventStream: Stream<Item = NotificationType<Block>> + Unpin,
+		EventStream: Stream<Item = NotificationType<Block>> + Unpin + Send,
 	{
-		let mut stream_item = stream.next();
+		// make the stream abortable
+		let (stream, handle) = futures::stream::abortable(stream);
+		let cancel_handle = handle.clone();
 
-		// The stop event can be triggered by the chainHead logic when the pinned
-		// block guarantee cannot be hold. Or when the client is disconnected.
-		let connection_closed = sink.closed();
-		tokio::pin!(connection_closed);
-		let mut stop_event = futures_util::future::select(rx_stop, connection_closed);
+		// create a oneshot to propagate error messages
+		let (tx_send, mut tx_receive) = oneshot::channel();
```

Comment:
Just use a mpsc channel instead?

We will probably just send message on it anyway but quite nice to get rid of the extra complexity with Mutex/swap stuff.